### ISSUE TITLE
Unable to enable class C in pre-activated/ABP device

### ISSUE THREAD

**Issue by sp193** [chirpstack/chirpstack#291]
I have a usecase, for transferring devices and their sessions between LNS (to/from Chirpstack). I have an adaptor that integrates Chirpstack into my software, which in turn can manage the devices within Chirpstack. So far, the device and its activation data can be transferred successfully, but it appears that the device is always created with only class A mode enabled.

The same thing can be observed, if you attempted to create a Class C device, in ABP mode.

The bottom entry shows a device which completed the OTAA join process, while the upper one shows a device that had its session data inserted into Chirpstack via gRPC.
```
chirpstack=# SELECT dev_eui,scheduler_run_after,enabled_class FROM device;
      dev_eui       |      scheduler_run_after      | enabled_class
--------------------+-------------------------------+---------------
 \x0095690e00000f56 |                               | A
 \x0095690e00000f57 | 2023-10-04 04:59:37.609275+00 | C
```

I guess, it is caused by set_enabled_class only getting called from join.rs? The Rust language is still unfamiliar to me.

**Comment by engbel** [comment]
Hi,
I think I'm experiencing the same situation.
Downlinks sent to devices operating as ABP/Class-C are waiting in the queue unless an uplink is sent from the device.
But downlinks sent to devices running in OTAA/Class-C mode are processed instantly.

**Comment by sp193** [comment]
Fixed in #344
